db/migration/migrations/000009_metadata_table.down.sql

```diff
@@ -0,0 +1,79 @@
+BEGIN;
+
+DROP VIEW IF EXISTS endpoint_export;
+
+DROP TRIGGER IF EXISTS update_fhir_endpoint_availability_trigger ON fhir_endpoints_info;
+DROP TRIGGER IF EXISTS update_fhir_endpoint_availability_trigger ON fhir_endpoints_metadata;
+DROP TRIGGER IF EXISTS set_timestamp_fhir_endpoints_metadata ON fhir_endpoints_metadata;
+
+ALTER TABLE fhir_endpoints_info 
+ADD COLUMN http_response INTEGER, 
+ADD COLUMN availability DECIMAL(5,4), 
+ADD COLUMN errors VARCHAR(500), 
+ADD COLUMN response_time_seconds DECIMAL(7,4), 
+ADD COLUMN smart_http_response INTEGER;
+
+ALTER TABLE fhir_endpoints_info_history 
+ADD COLUMN http_response INTEGER, 
+ADD COLUMN availability DECIMAL(5,4), 
+ADD COLUMN errors VARCHAR(500), 
+ADD COLUMN response_time_seconds DECIMAL(7,4), 
+ADD COLUMN smart_http_response INTEGER;
+
+ALTER TABLE fhir_endpoints_info
+DISABLE TRIGGER add_fhir_endpoint_info_history_trigger;
+
+
+CREATE OR REPLACE FUNCTION populate_existing_tables_endpoints_info() RETURNS VOID as $$
+    DECLARE
+        i RECORD;
+    BEGIN
+        FOR i IN SELECT DISTINCT fhir_endpoints_metadata.id, fhir_endpoints_metadata.http_response, fhir_endpoints_metadata.availability, fhir_endpoints_metadata.errors, fhir_endpoints_metadata.response_time_seconds, fhir_endpoints_metadata.smart_http_response FROM fhir_endpoints_metadata
+        LOOP
+            UPDATE fhir_endpoints_info SET http_response=i.http_response, availability=i.availability, errors=i.errors, response_time_seconds=i.response_time_seconds, smart_http_response=i.smart_http_response WHERE metadata_id = i.id;
+            UPDATE fhir_endpoints_info_history SET http_response=i.http_response, availability=i.availability, errors=i.errors, response_time_seconds=i.response_time_seconds, smart_http_response=i.smart_http_response WHERE metadata_id = i.id;
```

Comment:
I think this wouldn't get you the same database if you had old data that you migrated up and then migrated back down. There would be a bunch of data in the metadata table that wouldn't get added back to the history table since they have ids that are not associated with any of the history rows that exist? I'm not sure of an easy way to fix this, or if it really matters? 
I also used an old database file, ran the up migrate with no issues, then tried to run this and it seemed to get stuck? I think you said that you had a similar issue running a migrate file (though i think that was up?) so i was curious if you've experienced the same thing down.

Reply:
I tried fixing this the same way I did for the up migration, but it still seems to be getting stuck, not sure what might make this different in that the same fix will not work now, I will look more into it. Just because I am having a hard time understanding, why wouldn't this get you the same database if you had old data that you migrated up and then migrated back down? Each history row has a unique metadata_id that corresponds to information in the metadata table, as part of the up migration script is adding a metadata_id for each row of the history table. If I am missing something let me know, but this was my understanding!

Reply:
ah no you're totally right, i had confused myself! so nvm, this seems totally fine as long as it works!

Reply:
I'll have to look into why it is getting stuck tomorrow, but thank you for helping me make sure there isn't another issue with it!

Reply:
Migration down script is all set now, I added some indexes and it sped up the script a ton!